pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -244,8 +245,9 @@ func (a *index) podWorkloadBuilder(
 				Protocol: instancedWaypoint.DefaultBinding.Protocol,
 				Port:     instancedWaypoint.DefaultBinding.Port,
 			}
-		} else if waypoint := fetchWaypointForWorkload(ctx, waypoints, namespaces, p.ObjectMeta); waypoint != nil {
+		} else if waypoint, err := fetchWaypointForWorkload(ctx, waypoints, namespaces, p.ObjectMeta); err == nil {
 			// there is a workload-attached waypoint, point there with a GatewayAddress
+			// TODO: report waypoint error
```

Comment:
what does it mean to report error? It's not clear to me. What would this entail?

Reply:
This is for pod/workload entry. Today we only report Service/SE. Ill make the comment more clear

Reply:
Ah ok yeah, thanks for clarifying.